### DISCUSSION TITLE
Exploring CVC5 for prototyping a relational database with complete relations

### DISCUSSION THREAD

**Discussion by DavidPratten** [cvc5/cvc5#10284] [General]
Hi all,

My name is David, and I'm a PhD student at the University of Technology, Sydney. 

I'm focused on minimising the relational database programmers' need to leave their declarative data transformation environment to encode business rules. My thesis is that the relational database may be usefully generalised by including complete relations. For an intro, see "Relational expressions for data transformation and computation". https://doi.org/10.1007/978-3-031-47843-7_17 

If this area of research also interests you, I'd love to hear from you.  https://profiles.uts.edu.au/student_David.R.Pratten

Anyway, I'm excited to explore CVC5 as a research tool.

All the best,

David

**Comment by martin-cs** [comment]
Hi @DavidPratten thanks for getting in touch.  I take it you know about...

https://cvc5.github.io/docs/cvc5-1.1.0/theories/sets-and-relations.html
https://cvc5.github.io/docs/cvc5-1.1.0/theories/bags.html

I'm not working on this area but I will see if I can find someone who is.
